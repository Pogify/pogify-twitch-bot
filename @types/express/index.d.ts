declare namespace Express {
  interface Request {
    twitch: {
      broadcaster_type: BroadecasterType;
      description: string;
      display_name: string;
      email?: string;
      id: string;
      login: string;
      offline_image_url: string;
      profile_image_url: string;
      type: type;
      view_count: number;
      created_at: string;
    };
  }
}

type BroadecasterType = "partner" | "affiliate" | "";
type type = "staff" | "admin" | "global_mod" | "";
