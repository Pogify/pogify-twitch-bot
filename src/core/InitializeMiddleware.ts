import { Express } from "express";
import CommonMiddleware from "../middleware/CommonMiddleware";

export default class InitializeMiddleware {
  public static InitializeCommonMiddleware(app: Express): void {
    const commonMiddleware = new CommonMiddleware(app);

    commonMiddleware.useCookieParser();
    commonMiddleware.useCors();
    commonMiddleware.useStaticFolder();
  }
}
