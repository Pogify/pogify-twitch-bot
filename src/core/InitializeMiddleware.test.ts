import express from "express";
import InitializeMiddleware from "./InitializeMiddleware";

test("test bind middleware", () => {
  const app = express();

  InitializeMiddleware.InitializeCommonMiddleware(app);

  expect(app._router.stack).toMatchSnapshot();
});
