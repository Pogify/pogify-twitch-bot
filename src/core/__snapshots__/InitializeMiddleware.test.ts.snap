// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test bind middleware 1`] = `
Array [
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "query",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "expressInit",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "cookieParser",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "corsMiddleware",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "serveStatic",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/public\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
]
`;
